haystack/logging.py

```diff
@@ -60,10 +61,10 @@ def configure_logging(use_json: Optional[bool] = None) -> None:
         use_json_env_var = os.getenv(HAYSTACK_LOGGING_USE_JSON_ENV_VAR)
         if use_json_env_var is None:
             # Automatically enable JSON logging if stderr is not a TTY
-            use_json = not sys.stderr.isatty()
+            use_json = not (sys.stderr.isatty() or hasattr(builtins, "__IPYTHON__"))
```

Comment:
If we need to detect we're on a notebook with have a function for that in `utils.jupyter`. 👀

Reply:
aaah, perfect. Will use all three checks then 😁

Reply:
done